src/extern/driver.f90

```diff
@@ -117,11 +120,27 @@ subroutine singlepoint(self, env, mol, chk, printlevel, restart, &
     !$omp critical (turbo_lock)
     inquire (file='gradient', exist=exist)
     if (exist) then
+      ! ### only TM output is supported for now ###
       call rdtm(env,mol%n, .true., energy, gradient, xyz_cached)
       cache = all(abs(xyz_cached - mol%xyz) < 1.e-10_wp)
+      if (cache) then
+        write(env%unit,'(/,a,/)') &
+         "Geometry is equivalent to the one in &
+         'gradient'. Reading gradient from: 'gradient'."
+      endif
     end if
     if (.not. cache) then
-      call wrtm(mol%n, mol%at, mol%xyz)
+      call generateFileName(tmpname, 'xtbopt', extension, mol%ftype)
```

Comment:
We use `xtbopt` for the final optimized structure, but this routine can be called from a single point or hessian calculation. Better use a more neutral filename here.